### DISCUSSION TITLE
Input autocomplete suggestion?

### DISCUSSION THREAD

**Discussion by michaelmoreno** [Textualize/textual#1232] [Q&A] [answered]
Looked in the docs, don't see anything. Is text autocomplete suggestions a feature?

**Comment by davep** [comment, marked as answer]
There isn't currently anything to provide that in Textual itself.

**Reply by Dragon1573** [reply]
I'm new to Textual and currently studying the tutorial using VSCode inside WSL2 (Ubuntu).

In [Writing textual CSS](https://textual.textualize.io/tutorial/#writing-textual-css) section, VSCode is regarding `stopwatch03.css` as browser CSS and jumps MANY syntax errors.

![image](https://user-images.githubusercontent.com/49941141/208284214-866aa48f-c5f9-411c-b89c-b1ea34110979.png)

Is there any VSCode extensions or special configurations for me (and other IDE/Editor users) to switch the syntax of `*.css` from browser standard dialect to Textual specific one?

> JavaFX also use a "language specific dialect" of CSS called "FXML-CSS".
>
> When writing this kind of CSS file in JetBrains IntelliJ IDEA, IntelliJ will detect and switch to this dialect for us. IntelliLang system of IntelliJ will prompt out some special labels & properties other than standard browser one.

**Reply by davep** [reply]
To the best of my knowledge nobody has produced such an extension for VSCode yet.

**Reply by Dragon1573** [reply]
Okay (T＿T)

What about JetBrains PyCharm or any other programs? Is there a program that support this dialect directly after `textual[dev]` successfully installed (or can add supported with some configurations)?

There's a JSON Schema system for JSON & YAML, configuring syntax and sometimes providing auto-completion support with IDEs or code editors. Does CSS support something like that for "syntax switching"? If it does, how can we use it?

**Reply by davep** [reply]
I'm not aware of anyone having done any such work for any editor or IDE so far. A couple of people have been looking at the idea of doing something for VSCode, but nobody has announced anything yet.

There's nothing that comes with `textual[dev]` for this.

**Comment by mharig** [comment]
There is darrenburns/textual-autocomplete.

Didn't us it, yet.
